integrated_channels/blackboard/client.py

```diff
@@ -57,17 +59,40 @@ def create_content_metadata(self, serialized_data):
 
         external_id = channel_metadata_item.get('externalId')
 
+        # Retrieve and pop the data not needed for course creation and necessary for content customization
+        enrollment_link_metadata = channel_metadata_item.pop('enrollment_metadata')
+        bb_course_description = enrollment_link_metadata.get('element')
+        course_content_child_data = self.generate_integration_content_data(
+            channel_metadata_item.get('name'),
+            enrollment_link_metadata.get('link'),
+            channel_metadata_item.pop('syllabus_body'),
+            channel_metadata_item.pop('image_url')
+        )
+
         # blackboard does not support all characters in our courseIds so let's gen a hash instead
         course_id_generated = self.generate_blackboard_course_id(external_id)
 
         copy_of_channel_metadata = copy.deepcopy(channel_metadata_item)
         copy_of_channel_metadata['courseId'] = course_id_generated
+        copy_of_channel_metadata['description'] = bb_course_description
 
         LOGGER.info("Creating course with courseId: %s", external_id)
         self._create_session()
         create_url = self.generate_course_create_url()
         response = self._post(create_url, copy_of_channel_metadata)
-        return response.status_code, response.text
+
+        bb_course_id = response.json().get('id')
+
+        LOGGER.info("Creating content page for Blackboard course with course ID={}".format(external_id))
+        bb_content_id = self.create_integration_content_for_course(bb_course_id, course_content_child_data)
+
+        success_body = 'Successfully created Blackboard integration course={bb_course_id} with integration ' \
+                       'content={bb_content_id}'.format(
+                           bb_course_id=bb_course_id,
+                           bb_content_id=bb_content_id,
+                       )
+
+        return HTTPStatus.OK.value, success_body
```

Comment:
why are you assuming success here? We aren't doing error handling here or in the create_integration_content_for_course method so right now if it gets a KeyError or ValueError or something because the value doesn't exist, it's just going to fail badly.

Reply:
Error handling is __mostly__ handled by the client's wrapped request methods (get, post, patch) which handle cases when BB returns anything other than a `200`. Since the majority of the calls here are posts (so 404's are _unlikely_), and I'm fairly certain BB doesn't pull a moodle and return 200's on failed requests, I think this is very unlikely to explode. Having said that, sanity checks don't hurt and who knows what edge cases may pop up in the future so I'm willing to add. 👍